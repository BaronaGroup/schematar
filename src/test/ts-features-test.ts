import { testTypescriptInterface } from './test-utils'

describe('ts-features', function () {
  it(
    'allowNull',
    testTypescriptInterface({ fields: { field: { type: String, allowNull: true, optional: true } } }, (tsInterface) => {
      expect(tsInterface).toMatchSnapshot()
    })
  )

  const simpleSchema = { fields: { field: String } }
  it(
    'omitExtraExports (false)',
    testTypescriptInterface(
      simpleSchema,
      (tsInterface) => {
        expect(tsInterface).toMatchSnapshot()
      },
      { omitExtraExports: false }
    )
  )

  it(
    'omitExtraExports (true)',
    testTypescriptInterface(
      simpleSchema,
      (tsInterface) => {
        expect(tsInterface).toMatchSnapshot()
      },
      { omitExtraExports: true }
    )
  )

  it(
    'exportHash',
    testTypescriptInterface(
      simpleSchema,
      (tsInterface) => {
        expect(tsInterface).toMatchSnapshot()
      },
      { exportHash: 'hashField' }
    )
  )

  it(
    'doNotImportObjectId',
    testTypescriptInterface(
      simpleSchema,
      (tsInterface) => {
        // Note: meant to be used internally
        expect(tsInterface).toMatchSnapshot()
      },
      { doNotImportObjectId: true }
    )
  )
})
