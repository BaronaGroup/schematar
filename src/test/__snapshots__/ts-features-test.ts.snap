// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-features allowNull 1`] = `
"// Generated file, do not edit!

// tslint:disable array-type
// @ts-ignore -- ignore possibly unused type parameters
export interface TestInterfaceBase<IDType, DateType> {
  field?: string | null
}"
`;

exports[`ts-features doNotImportObjectId 1`] = `
"// Generated file, do not edit!

// tslint:disable array-type
// @ts-ignore -- ignore possibly unused type parameters
export interface TestInterfaceBase<IDType, DateType> {
  field: string
}
export type TestInterfaceMongoose = TestInterfaceBase<ObjectId, Date>
export type TestInterfaceJSON = TestInterfaceBase<string, string>
export type TestInterfaceFluid = TestInterfaceBase<string | ObjectId, string | Date>"
`;

exports[`ts-features exportHash 1`] = `
"// Generated file, do not edit!

// tslint:disable array-type
// @ts-ignore -- ignore possibly unused type parameters
export interface TestInterfaceBase<IDType, DateType> {
  field: string
}
import {ObjectId} from 'mongodb'
export type TestInterfaceMongoose = TestInterfaceBase<ObjectId, Date>
export type TestInterfaceJSON = TestInterfaceBase<string, string>
export type TestInterfaceFluid = TestInterfaceBase<string | ObjectId, string | Date>
export const hashField = '5d3861b32891d91c94cafb81aa1f821ae442e516d406e1e3bea47f6cb312b11f'"
`;

exports[`ts-features omitExtraExports (false) 1`] = `
"// Generated file, do not edit!

// tslint:disable array-type
// @ts-ignore -- ignore possibly unused type parameters
export interface TestInterfaceBase<IDType, DateType> {
  field: string
}
import {ObjectId} from 'mongodb'
export type TestInterfaceMongoose = TestInterfaceBase<ObjectId, Date>
export type TestInterfaceJSON = TestInterfaceBase<string, string>
export type TestInterfaceFluid = TestInterfaceBase<string | ObjectId, string | Date>"
`;

exports[`ts-features omitExtraExports (true) 1`] = `
"// Generated file, do not edit!

// tslint:disable array-type
// @ts-ignore -- ignore possibly unused type parameters
export interface TestInterfaceBase<IDType, DateType> {
  field: string
}"
`;
